Setup page after installation (#115)
vmware/precaution: docs/initial_setup.md

<!--
    Copyright 2019 VMware, Inc.
    SPDX-License-Identifier: BSD-2-Clause
-->

# Initial setup

> ## Run static code analysis locally (recommended)

Reviewer: ```suggestion
## Run static code analysis tooling (recommended)
```

> In order to achieve the best results when using Precaution, it's important to run static code analysis on the codebase in your main branch locally. 

Reviewer: This paragraph still needs to be repharased.

> Precaution scans new pull requests for security vulnerabilities and if in a pull request there are modifications of old source code it is likely that Precaution will find issues not related with the pull request changes but with the old source code in the modified file.

Reviewer: Probably remove line 10

The reason to do that is that there can be many older source code files which are stable and won't be changed soon by a pull request but still it's good to ensure they don't contain common security-related mistakes.

> You can run static code analysis for source code in Python using Bandit: https://github.com/PyCQA/bandit 

Reviewer: ```suggestion
You can run static code analysis of Python source code using Bandit: https://github.com/PyCQA/bandit 
```

> Therefore for best results, before using Precaution it is recommended to run static code analysis for security problems locally. That way when you resolve the possible issues you can ensure that the current code base does not contain common security issues.

Reviewer: ```suggestion
For best results, prior to using Precaution, it is recommended to run static code analysis tooling on the code base. That way the current code base does not contain known security issues.
```

> and for source code in Go using Gosec: https://github.com/securego/gosec 

Reviewer: ```suggestion
and for source code in Go use Gosec: https://github.com/securego/gosec 
```

## Setting up a Branch Protection Rule

By default, after Precaution is installed, it will not automatically prevent the merging of a pull request even if the check resulted in a failure status. 

> ![check_fails](./images/check_fails.png)

Reviewer: Please update this image now that Precaution is not using my profile as a logo.

Author: I will.

In order to allow Precaution to prevent pull requests from merging, do the following:

### 1. Create a pull request

You should create a pull request after the installation of Precaution in order to setup Precaution as a branch protection rule.

This pull request doesn't have to be merged.

### 2. Go to settings

![settings](./images/settings.png)

### 3. Choose branches

![branches](./images/branches.png)

### 4. Create a new branch protection rule 

![add_rule](./images/add_rule.png)

### 5. Setup the branch protection rule 

If you want to apply this rule to the master branch you will have to check the following options:

![branch_protection_rule.png](./images/branch_protection_rule.png)

> Congratulations! Branch protection is now enabled and will prevent merging code that fails the status check.

Reviewer: This is redundant.  Branch protection and "prevent merging" are the same.

Author: I changed it a little. Can you have a look?
